libraries/ESP8266WiFi/src/WiFiClientSecure.h

```diff
@@ -34,6 +34,8 @@ class WiFiClientSecure : public WiFiClient {
   ~WiFiClientSecure() override;
   WiFiClientSecure(const WiFiClientSecure&);
   WiFiClientSecure& operator=(const WiFiClientSecure&);
+  // Only called by WiFiServerSecure
+  WiFiClientSecure(ClientContext* client, bool usePMEM, const uint8_t *rsakey, int rsakeyLen, const uint8_t *cert, int certLen);
```

Comment:
Perhaps can be made `protected` and then WiFiServerSecure can be made `friend`?

Reply:
Done.

I had a friend class originally, but a reviewer suggested that, as per the C++ FAQ, that "friend class" is very much frowned upon by the C++ community.  Being a hardware person, I took the FAQ at its word.